.github/workflows/python-tests.yml

```
@@ -0,0 +1,57 @@
+name: Python Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
```

Run test cases only when pr is raised to `main` branch
```suggestion
     branches:
          - main

```